apps/sms/test/marionette/new_activity_test.js

```diff
@@ -18,11 +18,45 @@ marionette('Messages as "new" activity target', function() {
   var messagesApp,
       activityCallerApp;
 
+  function launchAsActivity(activityData) {
+    var smsMessage = {
+      id: 1,
+      iccId: null,
+      threadId: 1,
+      sender: null,
+      receiver: '+200',
+      type: 'sms',
+      delivery: 'sent',
+      body: 'Workload message',
+      timestamp: Date.now()
+    };
+
+    activityCallerApp.sendMessage(activityData);
+    messagesApp.switchTo();
+    messagesApp.setStorage([{
+      id: smsMessage.threadId,
```

Comment:
You don't need to think too hard, both approaches are good, so either is good :) I like the finer-grained one with several calls.